README.MD

```diff
@@ -166,40 +173,86 @@ struct BatchedOutputOrders {
 }
 ```
 
+#### Example `processInputOrders` flow
+
+<div align="center"><img src="./static/processInputOrders.png" width="1200"></div>
+
+## Nested Factory interaction with the Nested Reserve and Nested Records
+The Nested Reserve stores underlying assets of all NestedNFTs. The Nested Records keeps track of which underlying assets are associated with a specific NestedNFT. 
+Hence, each time the Nested Factory needs to interact with user funds (which are represented as a NestedNFT), it will first check the balance of tokens associated with the NestedNFT through Nested Records. If needed, it will then transfert funds to the Nested Reserve or withdraw funds from it.
+
+## ETH managment
+
+The Nested protocol only handles WETH when calling operators.  
+The sent ETH used to feed a portfolio is automatically converted to WETH when received.
+
+### WETH Conversion
+
+The conversion from ETH to WETH is done when submitting an order through `_submitInOrders` or `_submitOutOrders`.
+
+Before submitting orders, the NestedFactory transfers the input tokens from the NestedReserve (or the msg.sender) to the factory and converts the sent ETH to WETH.
+
+### Use of ETH in operators
+
+There are some operators who use ETH directly and not WETH.  
+In this case, the operator uses the Withdrawer to get the ETH back from the WETH contract before using it.
+
+> Only the Withdrawer can send ETH to the NestedFactory without automatic WETH conversion.
+
+## Royalties
+Royalties are withdrawal fees when a user has copied an existing portfolio and are managed by the `FeeSplitter.sol`. Royalties are distributed from the `NestedFactory.sol` when a user withdraws funds from his portfolio OR when an order fails, all tokens are returned to the sender and are therefore subject to withdrawal fees.
```

Comment:
```suggestion
Royalties are withdrawal fees allocated to the owner of the original portfolio when a user has copied an existing portfolio. They are managed by the `FeeSplitter.sol`. Royalties are distributed from the `NestedFactory.sol` when a user withdraws funds from his portfolio OR when an order fails, all tokens are returned to the sender and are therefore subject to withdrawal fees.
```

Reply:
Royalties are a part of the fee collected by the protocol.
Thus royalties are collected during every step of the copied portfolio lifecycle (copy, update, deposit, withdraw) not only upon withdrawing funds